devito/operator.py

```diff
@@ -207,6 +214,11 @@ def apply(self, debug=False):
             if hasattr(param, 'initialize'):
                 param.initialize()
         args = [param.data for param in self.signature]
+
+        # if auto tuning block sizes return compiled function and its arguments to AutoTuner
+        if auto_tune:
+            return f, args
```

Comment:
This hot-wiring is really not great! Is this only required to feed your own custom timer functions, or is there a deeper reason?